frontend/src/metabase/hooks/use-synced-query-string.ts

```diff
@@ -26,23 +27,23 @@ export function useSyncedQueryString(object: Record<string, any>) {
         window.location.pathname + searchString + window.location.hash,
       );
     }
+  }, [object]);
 
-    return () => {
-      // Remove every previously-synced keys from the query string when the component is unmounted.
-      // This is a workaround to clear the parameter list state when [SyncedParametersList] unmounts.
-      const searchString = buildSearchString({
-        filterFn: key => !(key in object),
-      });
+  useUnmount(() => {
+    // Remove every previously-synced keys from the query string when the component is unmounted.
+    // This is a workaround to clear the parameter list state when [SyncedParametersList] unmounts.
+    const searchString = buildSearchString({
+      filterFn: key => !(key in object),
```

Comment:
This looks incorrect.

> Remove every previously-synced keys

At this point we don't know how `object` changed over time so we can't reliably clean up after it. We only have access to currently-synced keys.

----

I think it should be enough to change `useEffect` to `useDeepCompareEffect` to fix the flickering.

Reply:
Off course you are right. I've updated the PR to use `useDeepCompareEffect`.